Framework/API/inc/MantidAPI/AlgorithmManager.h

```diff
@@ -44,6 +44,8 @@ class MANTID_API_DLL AlgorithmManagerImpl {
   IAlgorithm_sptr create(const std::string &algName, const int &version = -1);
   /// Creates an unmanaged algorithm with the option of choosing a version
   std::shared_ptr<Algorithm> createUnmanaged(const std::string &algName, const int &version = -1) const;
+  /// Creates a managed algorithm with the option of choosing a version (used by python interface)
+  IAlgorithm_sptr createFromPython(const std::string &algName, const int &version = -1);
```

Comment:
Would calling this `createGILSafe` or something similar provide a better description of why we want it to only be called from the python interface?

Reply:
Yeah this is a better name